graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -484,6 +495,309 @@ auto GetTemporalPickFn(
+      etype_id_to_num_picked_offset[id + 1] =
+          seed_offsets->at(dst_ntype_id + 1) - seed_offsets->at(dst_ntype_id) +
+          1;
+    }
+    for (int64_t i = 1; i <= num_etypes; ++i) {
+      etype_id_to_num_picked_offset[i] += etype_id_to_num_picked_offset[i - 1];
+    }
+  } else {
+    etype_id_to_dst_ntype_id[0] = 0;
+    etype_id_to_num_picked_offset[1] = num_seeds + 1;
+  }
+  // `num_rows` indicates the length of `num_picked_neighbors_per_node`, which
+  // is used for storing pick numbers. In non-temporal hetero sampling, it
+  // equals to sum_{etype} #seeds with ntype=dst_type(etype). In homo sampling,
+  // it equals to `num_seeds`.
+  const int64_t num_rows = (fanouts.size() == 1 || !with_seed_offsets)
+                               ? num_seeds + 1
+                               : etype_id_to_num_picked_offset[num_etypes];
+  torch::Tensor num_picked_neighbors_per_node =
+      torch::zeros({num_rows}, indptr_options);
+
+  AT_DISPATCH_INDEX_TYPES(
+      indptr_.scalar_type(), "SampleNeighborsImplWrappedWithIndptr", ([&] {
+        using indptr_t = index_t;
+        AT_DISPATCH_INDEX_TYPES(
+            seeds.scalar_type(), "SampleNeighborsImplWrappedWithSeeds", ([&] {
+              using seeds_t = index_t;
+              const auto indptr_data = indptr_.data_ptr<indptr_t>();
+              const auto num_picked_neighbors_data_ptr =
+                  num_picked_neighbors_per_node.data_ptr<indptr_t>();
+              num_picked_neighbors_data_ptr[0] = 0;
+              const auto seeds_data_ptr = seeds.data_ptr<seeds_t>();
+
+              // Step 1. Calculate pick number of each node.
+              torch::parallel_for(
+                  0, num_seeds, grain_size, [&](int64_t begin, int64_t end) {
+                    for (int64_t i = begin; i < end; ++i) {
+                      const auto nid = seeds_data_ptr[i];
+                      TORCH_CHECK(
+                          nid >= 0 && nid < NumNodes(),
+                          "The seed nodes' IDs should fall within the range of "
+                          "the graph's node IDs.");
+                      const auto offset = indptr_data[nid];
+                      const auto num_neighbors = indptr_data[nid + 1] - offset;
+
+                      if (with_seed_offsets) {
+                        const auto seed_type_id =
+                            (fanouts.size() > 1) ? std::upper_bound(
+                                                       seed_offsets->begin(),
+                                                       seed_offsets->end(), i) -
+                                                       seed_offsets->begin() - 1
+                                                 : 0;
+                        // `seed_offset` indicates the index of the current
+                        // seed within the group of seeds which have the same
+                        // node type.
+                        const auto seed_offset =
```

Comment:
A better name would be more appropriate, right now it is same as the seed_offsets tensor.

Reply:
Renamed to `seed_index`